src/core/raster/qgsrasterminmaxorigin.h

```diff
@@ -0,0 +1,156 @@
+/***************************************************************************
+    qgsrasterminmaxorigin.h - Origin of min/max values
+     --------------------------------------
+    Date                 : Dec 2016
+    Copyright            : (C) 2016 by Even Rouault
+    email                : even.rouault at spatialys.com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSRASTERMINMAXORIGIN_H
+#define QGSRASTERMINMAXORIGIN_H
+
+#include <QDomDocument>
+#include <QDomElement>
+
+/** \ingroup core
+ * This class describes the origin of min/max values. It does not store by
+ * itself the min/max values.
+ * @note added in QGIS 3.0
+ */
+class CORE_EXPORT QgsRasterMinMaxOrigin
+{
+public:
+
+    //! \brief Default cumulative cut lower limit
+    static const double CUMULATIVE_CUT_LOWER;
```

Comment:
What do you think about flipping these static consts to `static constexpr double`?